llvm/lib/Target/AArch64/AArch64Combine.td

```diff
@@ -290,6 +290,13 @@ def combine_mul_cmlt : GICombineRule<
   (apply [{ applyCombineMulCMLT(*${root}, MRI, B, ${matchinfo}); }])
 >;
 
+def lower_build_insert_vec_elt : GICombineRule<
```

Comment:
Bump on this, why is this an optimization combine and not in LegalizerHelper? The title says its lowering, but this is not

Reply:
This doesn't seem good. Unrolling a build_vector into a sequence of vector inserts naturally belongs in LegalizerHelper

Reply:
I'm not sure you can get through legalization without buildvector (or merge), without it making quite a mess. This fits in with the existing way that AArch64 handles shuffle vectors / buildvectors / other vector operations. It has lower in the name but it can be thought of as a combine that splits the buildvectors up prior to selection to help use the existing tablegen patterns.